JetStream3 harness: reset seed of Math.random() on every benchmark iteration

shvaikalesh opened the pull request:
#### 7bca60ef1a1704e00bd8ca90c006793f0556cb0d
<pre>
JetStream3 harness: reset seed of Math.random() on every benchmark iteration
<a href="https://bugs.webkit.org/show_bug.cgi?id=249592">https://bugs.webkit.org/show_bug.cgi?id=249592</a>

Reviewed by Yusuke Suzuki.

This change extracts Math.random() implementation, which resets its seed on every iteration,
from simple/file-system.js to be used for all other tests, including newly-added BigInt crypto
tests that heavily rely on Math.random().

* PerformanceTests/JetStream3/JetStreamDriver.js:
(prototype.get runnerCode):
(prototype.get preiterationCode):
(prototype.async run):
(AsyncBenchmark.get runnerCode.return.async doRun):
(AsyncBenchmark.prototype.get runnerCode):
(AsyncBenchmark):
* PerformanceTests/JetStream3/simple/file-system.js:
(Benchmark.prototype.async runIteration):
(Benchmark):
(resetSeed): Deleted.
(Math.random): Deleted.

Canonical link: <a href="https://commits.webkit.org/258213@main">https://commits.webkit.org/258213@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e35def308d6280321e611c96c1fb7c3e401ef49

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100889 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10042 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33938 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110196 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10973 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/911 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93304 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108038 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106672 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8297 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91555 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34911 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90205 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22952 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/77886 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3730 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24472 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3748 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/868 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9861 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43968 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5526 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2963 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (changes requested):
I think we should just use this deterministic random thing for crypto test too since `__resetSeed` is not a standard function.
V8 / SpiderMonkey will not support it, and exposing different results due to non-standard function is not OK as a performance benchmark test.

Constellation reviewed (approved):
Oops, sorry. I completely misunderstood the patch!! (I wasm thinking about the opposite thing). This patch is applying the deterministic random to the other benchmarks. This is totally making sense.

Commit:
JetStream3 harness: reset seed of Math.random() on every benchmark iteration
https://bugs.webkit.org/show_bug.cgi?id=249592

Reviewed by Yusuke Suzuki.

This change extracts Math.random() implementation, which resets its seed on every iteration,
from simple/file-system.js to be used for all other tests, including newly-added BigInt crypto
tests that heavily rely on Math.random().

* PerformanceTests/JetStream3/JetStreamDriver.js:
(prototype.get runnerCode):
(prototype.get preiterationCode):
(prototype.async run):
(AsyncBenchmark.get runnerCode.return.async doRun):
(AsyncBenchmark.prototype.get runnerCode):
(AsyncBenchmark):
* PerformanceTests/JetStream3/simple/file-system.js:
(Benchmark.prototype.async runIteration):
(Benchmark):
(resetSeed): Deleted.
(Math.random): Deleted.

Canonical link: https://commits.webkit.org/258213@main
